### ISSUE TITLE
setValues retrieves old state in the function variant

### ISSUE THREAD

**Issue by KyorCode** [formium/formik#3047]
### Current Behavior

setValues has retrieves old state in the function variant. setValues(prev => .... )

### Expected behavior

That setValues behaves the same as react his setState .

### Reproducible example

https://codesandbox.io/s/vigorous-sky-ttcj1?file=/src/App.js

Adjust x to an other number ( as long as the change triggers ), I expect z to be increased by 2, not 1.

Compared to a setState implementation, set setState call always get the latest state.

https://codesandbox.io/s/tender-framework-wk87g?file=/src/App.js

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           | 2.2.6
| React            | 17.0.1
| TypeScript       | 
| Browser          | 
| npm/Yarn         |
| Operating System |

**Comment by KyorCode** [comment]
Shameless bump after 27days of nothing...

**Comment by johnrom** [comment]
This is true, in order to do what you're looking for the function would have to be passed into the reducer and called with the state inside the reducer.

**Comment by KyorCode** [comment]
Sadly still not addressed

**Comment by johnrom** [comment]
It is fixed in #3231 which is... just sitting there.

**Comment by KyorCode** [comment]
Its sad that it seems like a core function, doesn't work as expected... even after it being a bug for more then 2 years...
